### ISSUE TITLE
[Feature]: Keep items (shows) that are continuing. 

### ISSUE THREAD

**Issue by JamsRepos** [shemanaev/jellyfin-plugin-media-cleaner#40]
### Describe the feature you'd like added

For example; The Boys isn't currently airing but a new season is due soon. Unless I set the days to something high say a year, this show will be deleted. So when the new show airs. it won't be put onto the server.

In an ideal world, there should be a checkbox to state whether to include continuing shows so you can have your days set to 90 for example (so everything gets deleted after 3 months if it's old content) but because it's got new episodes next year, it will remain.

**Comment by JamsRepos** [comment]
It's also worth mentioning this data already exists within a shows metadata, forgot to include that in my original post. So this should be somewhat, of an easy thing to do without the use of radarr/sonarr API access.

**Comment by shemanaev** [comment]
Hi, have you tried "Delete after played" => "Entire series has ended"? It has some limitations though https://github.com/shemanaev/jellyfin-plugin-media-cleaner/issues/5#issuecomment-1243810645

**Comment by JamsRepos** [comment]
Apologies for not seeing this, will this only delete if the whole entire season has been watched AND it's marked as Ended?

**Comment by shemanaev** [comment]
> entire season has been watched AND it's marked as Ended

Series, not season. Otherwise it _should_ work that way.

**Comment by JamsRepos** [comment]
I see, I think this is perhaps limited like you say, as if for example The Simpsons has 36 Seasons. Obviously it would take a long time for someone to get through the whole entire lot and then have to wait for them to obviously stop airing it (which is fairly unlikely). We'd only really want to keep the last season if it's still currently airing.

**Comment by shemanaev** [comment]
> We'd only really want to keep the last season if it's still currently airing.

Yeah, I'll look if the same logic can be applied to the seasons.

**Comment by JamsRepos** [comment]
> > We'd only really want to keep the last season if it's still currently airing.
> 
> Yeah, I'll look if the same logic can be applied to the seasons.

Same logic meaning? If a season is continuing? As if this is the case then we'd still have the issue where a show is deleted and then when the new season airs it won't display in the continue watching when it eventually downloads and it would also be unmonitored in the arr.

**Comment by JamsRepos** [comment]
> *arr relies on the physical presence of files/folders?

Yes - but there is an option to unmonitor if the files is deleted (so then a service like overseerr/jellyseerr can then detect them as no longer available and allow them to be requested again which is almost a must-have feature with this plugin otherwise sonarr/radarr would just re-download content once it's missing which would defeat the purpose.

> Keep last season of the show even it was played? Keep only the last episode of the show like in https://github.com/shemanaev/jellyfin-plugin-media-cleaner/issues/14?

I feel like both work - although leaving the last episode of the show may look a little odd. I'd say season is probably best as it's then able to be used as a "catch-up" by users if they want to watch it again prior to the new season launching or whatever. Plus, it's still better than leaving all seasons there. 

I feel like in a best case scenario, both would be an option but that's up to you if you'd like to give that granularity, if not then we can always discuss as a group what would best be suited.

**Comment by shemanaev** [comment]
> Yes - but there is an option to unmonitor if the files is deleted (so then a service like overseerr/jellyseerr can then detect them as no longer available and allow them to be requested again which is almost a must-have feature with this plugin otherwise sonarr/radarr would just re-download content once it's missing which would defeat the purpose.

Ah, got it.


> I feel like both work - although leaving the last episode of the show may look a little odd. I'd say season is probably best as it's then able to be used as a "catch-up" by users if they want to watch it again prior to the new season launching or whatever. Plus, it's still better than leaving all seasons there.

Yes, this makes sense. 

Conditions are clear now 😄 



> I feel like in a best case scenario, both would be an option but that's up to you if you'd like to give that granularity, if not then we can always discuss as a group what would best be suited.

I need to research what will be best/easier to implement and will report back.

**Comment by shemanaev** [comment]
@JamsRepos can you test [this build](https://github.com/shemanaev/jellyfin-plugin-media-cleaner/actions/runs/8124875173/artifacts/1291948304)? There is new options under "Delete after played" for series.

**Comment by JamsRepos** [comment]
> @JamsRepos can you test [this build](https://github.com/shemanaev/jellyfin-plugin-media-cleaner/actions/runs/8124875173/artifacts/1291948304)? There is new options under "Delete after played" for series.

Awesome - I've got it installed now and i'll let you know how it goes. Is there a way I can mark a show as last watched X amount of time ago to test? Otherwise I risk having things delete which I don't really want by setting the threshhold lower.

**Comment by shemanaev** [comment]
> Is there a way I can mark a show as last watched X amount of time ago to test?

I don't think there is a way other than editing database directly 😞

**Comment by JamsRepos** [comment]
> > Is there a way I can mark a show as last watched X amount of time ago to test?
> 
> I don't think there is a way other than editing database directly 😞

I can confirm that the epsiode one is working fine - only have the last episode in the last season of a Continuing show.

**Comment by JamsRepos** [comment]
Ok - correction. It doesn't seem to be. The Last Ship has deleted some episodes and others it doesn't. 
<img width="638" alt="image" src="https://github.com/shemanaev/jellyfin-plugin-media-cleaner/assets/1347620/c5ecc3ee-673f-4318-a1b7-680cc9bfb03d">

I can't really see watch history of them to assume that it's been watched.

Season 1 has Episode 8.
Season 2 has Episode 1 and Episode 8.

Might require you to do some of your own investigation cause i'm slightly confused as to what is happening. It's quite hard for me to access anything currently.
